### PR TITLE
Fixes #1186: Storage Cells drop upgrades upon disassembling

### LINKED ISSUE #1186: Dismantling an empty storage cell with a fuzzy card voids the card
Hi,

let me just list the steps to reproduce this bug:
1. Get any storage cell
2. Put it into a cell workbench
3. Install a fuzzy card into the cell
4. Take the cell out of the workbench
5. Shift-right click with the cell in your hand
6. You get a storage housing and a storage component, but the fuzzy card is gone.

This happens in version AE2 rv2-beta-8.

### PR DESCRIPTION
**Pull request by thatsIch**
only `ItemBasicStorageCell.java` contains the relevant code, rest was noise to detect the required code to fix the bug

### CODE REVIEW

File: src/api/java/appeng/api/implementations/items/IStorageCell.java [hunk 1]
```diff
@@ -59,10 +59,22 @@
...
 	int BytePerType( ItemStack cellItem );
 
 	/**
+	 * Determines the number of bytes used for any type included on the cell.
+	 *
+	 * @param cellItem item
+	 *
+	 * @return number of bytes
+	 */
+	int getBytePerType( ItemStack cellItem );
```
Comment (yueh) [inline src/api/java/appeng/api/implementations/items/IStorageCell.java]: Maybe bytes as it is never really singular?

**Comment by thatsIch** [comment]
Changed it

**Comment by yueh** [comment]
The cell and housing is directly placed in the player inventory and only dropped when not enough space is left. For example `ItemStack extraB = ia.addItems( this.component.stack( 1 ) );`.
Maybe also handle the upgrades in the same way?

**Comment by thatsIch** [comment]
Added placing item into inventory

---

It should not, its just marked as deprecated and I gave them already a headsup. Internally its still using the old `BytePerType`. So not implementing it, should not crash them

€: tested it, it does not break with their most current version by placing their cells into a ME Drive (since the only call is in `CellInventory`

**Pull request merged**